teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -72,8 +91,329 @@ Per [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-
 **Tekton Maintainers**
 - I would like to be able to migrate the apiVersion without having to make backwards incompatible changes.
 
+## Requirements
+- Avoid making changes to existing features' stability levels just for the purpose of addressing coupling feature and API versioning.
+- Avoid blocking the promotion of `beta` features from the existing `alpha` features.
+- It should have a testing stratey that will give us confidence in our impelmentations of per feature flags and changes to existing feature flags.
+
+## Proposal
+
+This TEP provides a plan for ensuring that feature stability doesn't depend on CRD API version, and using per-feature flags to move to a future where only stable features are enabled by default through the following steps:
+- [Change existing validation to decouple feature and API versioning](#change-existing-validation-to-decouple-feature-and-api-versioning)
+- [Per feature flag for new api-driven features](#per-feature-flag-for-new-api-driven-features)
+- [Sunset `enable-api-fields` after existing features stabilize](#sunset-enable-api-fields-after-existing-features-stabilize)
+
+
+## Design Details
+
+### Change existing validation to decouple feature and API versioning
+We will change the current validation for `enable-api-fields=stable` to only allow using stable features regardless of API version. This will resolve the current issue of the coupling of API and feature versioning in v1beta1. More specifically, beta features resolvers, object/ array params and results will require `enable-api-fields` set to `beta` to be used.
+
+Note that although this is a behavior change, it is more of a bug fix for the coupling of feature and API versioning. Currently, with `enable-api-fields` set to `stable`,  PipelineRuns like [this one](https://github.com/tektoncd/pipeline/blob/main/examples/v1/pipelineruns/beta/git-resolver.yaml) fail because the controller cannot create child TaskRuns. This change will result in a validation failure instead.
```

Comment:
Can we add something to give an understanding on what it means ? Something like :

This would mean that, today, to use resolvers, you need to set `enable-api-fields` to `false`, even if you are using `v1beta1` APIs (for `PipelineRun`, …). The default value being `beta`, it wouldn't break users *by default*, but it would change a behavior for those who only wanted to enable `stable` features.

Reply:
Added. Also we have included the impacts on users below for elaborating those use cases. Hope that helps)